test/e2e/common/fixtures.py

```diff
@@ -0,0 +1,110 @@
+# License is located at
+#
+# 	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+"""Common SageMaker test fixtures.
+"""
+
+import pytest
+
+from e2e import (
+    CONFIG_RESOURCE_PLURAL,
+    MODEL_RESOURCE_PLURAL,
+    ENDPOINT_RESOURCE_PLURAL,
+    DATA_QUALITY_JOB_DEFINITION_RESOURCE_PLURAL,
+    create_sagemaker_resource,
+    wait_sagemaker_endpoint_status,
+)
+
+from e2e.replacement_values import REPLACEMENT_VALUES
+from acktest.resources import random_suffix_name
+from acktest.k8s import resource as k8s
+
+
+@pytest.fixture(scope="module")
+def xgboost_churn_endpoint(sagemaker_client):
+    """Creates a SageMaker endpoint with the XGBoost churn single-variant model
+    and data capture enabled.
+    """
+    endpoint_resource_name = random_suffix_name("xgboost-churn", 32)
+    config_resource_name = endpoint_resource_name + "-config"
+    model_resource_name = config_resource_name + "-model"
+
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["ENDPOINT_NAME"] = endpoint_resource_name
+    replacements["CONFIG_NAME"] = config_resource_name
+    replacements["MODEL_NAME"] = model_resource_name
+    data_bucket = replacements["SAGEMAKER_DATA_BUCKET"]
+    replacements[
+        "MODEL_LOCATION"
+    ] = f"s3://{data_bucket}/sagemaker/model/xgb-churn-prediction-model.tar.gz"
+
+    model_reference, model_spec, model_resource = create_sagemaker_resource(
+        resource_plural=MODEL_RESOURCE_PLURAL,
+        resource_name=model_resource_name,
+        spec_file="xgboost_model_with_model_location",
+        replacements=replacements,
+    )
+    assert model_resource is not None
+    assert k8s.get_resource_arn(model_resource) is not None
+
+    config_reference, config_spec, config_resource = create_sagemaker_resource(
+        resource_plural=CONFIG_RESOURCE_PLURAL,
+        resource_name=config_resource_name,
+        spec_file="endpoint_config_data_capture_single_variant",
+        replacements=replacements,
+    )
+    assert config_resource is not None
+    assert k8s.get_resource_arn(config_resource) is not None
+
+    endpoint_reference, endpoint_spec, endpoint_resource = create_sagemaker_resource(
+        resource_plural=ENDPOINT_RESOURCE_PLURAL,
+        resource_name=endpoint_resource_name,
+        spec_file="endpoint_base",
+        replacements=replacements,
+    )
+    assert endpoint_resource is not None
+    assert k8s.get_resource_arn(endpoint_resource) is not None
+    wait_sagemaker_endpoint_status(
+        sagemaker_client, replacements["ENDPOINT_NAME"], "InService"
+    )
+
+    yield endpoint_spec
+
+    for cr in (model_reference, config_reference, endpoint_reference):
+        assert k8s.assert_delete_resource_successful(cr)
```

Comment:
I don't think the PR in test-infra that adds the `assert_delete_resource_successful` function will be merged. Not only does the `assert_delete_resource_successful` function not actually assert anything, but it's actually more readable to just write:

```python
_, deleted = k8s.delete_custom_resource(cr)
assert(deleted)
```

alternately, I'd be willing to entertain updating the `k8s.delete_custom_resource` function to just return `True` or `False` instead of a tuple of response object and boolean.